src/dmd/root/file.d

```diff
@@ -309,10 +299,20 @@ nothrow:
 
     /* Set buffer
      */
-    extern (C++) void setbuffer(void* buffer, size_t len)
+    extern (C++) void setbuffer(void* buffer, size_t len, bool transferOwnership = false)
```

Comment:
There are three functions here, that seem to be doing more or less the same thing:

1. setbuffer(buf,len,false)
2. setbuffer(buf,len,true)
3. hijackBufferFrom(buffer)

Writing them out like that, I have no idea what they do. This may sound like nitpicking, but the critical problem here is who owns the buffer. Getting it wrong means ugly bugs, so I'd like this to be 3 separate functions with clear names for what they do.

For example, `OutBuffer` has `peek` for looking at data and not taking control, while `extract` for taking the data away from the `OutBuffer`. After using those for a while, I haven't found it confusing and am pretty pleased with the convention. May I suggest:

1. setView() - sets the buffer, but does not transfer ownership
2. transferTo() - sets the buffer, and transfers ownership

I suggest abandoning hijackBufferFrom, replacing it with the pair extractData() and transferTo(). It should look like:
```
from.extractData().transferTo(to);
```